editoast/src/views/train_schedule/mod.rs

```diff
@@ -393,29 +403,88 @@ async fn create_backend_request_payload(
 ) -> Vec<SimulationOutputChangeset> {
     let mut simulation_outputs = Vec::new();
+    let electrification_conditions_iter = if electrification_conditions.is_empty() {
+        vec![None; base_simulations.len()]
+    } else {
+        electrification_conditions
+            .iter()
+            .map(|ec| Some(ec.clone()))
+            .collect()
+    };
     for (base_simulation, eco_simulation, speed_limits, electrification_conditions) in izip!(
-        response_payload.base_simulations,
-        response_payload.eco_simulations,
-        response_payload.speed_limits,
-        response_payload.electrification_conditions
+        base_simulations,
+        eco_simulations,
+        speed_limits,
+        electrification_conditions_iter
     ) {
+        let eco_simulation = match eco_simulation {
+            Some(eco) => Some(Some(DieselJson(eco))),
+            None => Some(None),
+        };
         let simulation_output = SimulationOutputChangeset {
             id: None,
             mrsp: Some(speed_limits),
             base_simulation: Some(DieselJson(base_simulation)),
-            eco_simulation: Some(Some(DieselJson(eco_simulation))),
-            electrification_conditions: Some(electrification_conditions),
-            train_schedule_id: None, // To be filled once the train schedule is inserted
+            eco_simulation,
+            electrification_conditions,
+            train_schedule_id: Some(None), // To be filled once the train schedule is inserted
         };
         simulation_outputs.push(simulation_output);
     }
     simulation_outputs
 }
 
+#[derive(Debug, QueryableByName)]
+pub struct OutputSimulationTrainSchedule {
+    #[diesel(sql_type = BigInt)]
+    pub id: i64,
+    #[diesel(sql_type = Jsonb)]
+    pub mrsp: JsonValue,
+    #[diesel(sql_type = Jsonb)]
+    pub base_simulation: DieselJson<ResultTrain>,
+    #[diesel(sql_type = Nullable<Jsonb>)]
+    pub eco_simulation: Option<DieselJson<ResultTrain>>,
+    #[diesel(sql_type = Jsonb)]
+    pub electrification_conditions: JsonValue,
+    #[diesel(sql_type = Nullable<BigInt>)]
+    pub train_schedule_id: Option<i64>,
+    #[diesel(sql_type = Nullable<Double>)]
+    pub schedule_departure_time: Option<f64>,
+}
+
+pub fn simulation_outputs_from_timetable_id(
+    given_timetable_id: i64,
+    db_pool: Data<DbPool>,
+) -> Result<Vec<OutputSimulationTrainSchedule>> {
+    let conn = &mut db_pool.get()?;
+    let result = sql_query(format!(
+        "SELECT 
+            s.id, 
+            s.mrsp, 
+            s.base_simulation, 
+            s.eco_simulation, 
+            s.electrification_conditions, 
+            s.train_schedule_id, 
+            t.departure_time as schedule_departure_time 
+        FROM osrd_infra_simulationoutput s
+        INNER JOIN osrd_infra_trainschedule t ON t.id = s.train_schedule_id
+        WHERE t.timetable_id = {};",
+        given_timetable_id
+    ))
+    .load::<OutputSimulationTrainSchedule>(conn)?;
+    Ok(result)
+}
+
```

Comment:
Same here, no unit tests (though i do agree that you didn't create this class, so it's really annoying to have to create everything from scratch...).

Reply:
not really the scope of the PR